Robot_Adapter/Read/Elements/Bar.cs

```diff
@@ -76,11 +77,11 @@ private List<Bar> ReadBars(List<string> ids = null)
                     bhomBars.Add(bhomBar);
                 }
             }
-            else if (ids != null && ids.Count > 0)
+            else if (barIds != null && barIds.Count > 0)
```

Comment:
We are checking exactly the same thing here now twice. This could be changed to an simple `else`